activerecord/lib/active_record/enum.rb

```diff
@@ -248,6 +255,10 @@ def define_enum_methods(name, value_method_name, value, scopes)
               klass.scope "not_#{value_method_name}", -> { where.not(name => value) }
             end
           end
+
+          def column_backed_by_string?(name)
+            klass.type_for_attribute(name).type == :string
```

Comment:
We can't call `type_for_attribute` here, since it will force the schema to load while the class is being defined.

Reply:
Do you have a different option?

Reply:
Maybe there’s a more elegant way, but what if the API was explicit?

enum cover: [:hard, :soft], _type: :text

Reply:
@ghiculescu I, in fact, considered this, but thought it was less cool.

It would look cool with new enum syntax:

```ruby
enum :cover, [:hard, :soft], type: :text
```

On the other hand, I have a feeling that the core team wants to encourage string backed enums, not integer backed ones. I wonder if this would make string backed enums look more 2nd class citizens.

@kaspth thoughts?

Reply:
This may have been lost i your notifications, so I'm pinging you again @kaspth

Reply:
@rafaelfranca do you have any thoughts here?